design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,349 @@
+
+### User Stories
+
+- As a cluster administrator, I want to configure static IPs on secondary NICs on a pool of nodes without creating one per each node.
+- As a cluster administrator, I want scale up the cluster with new nodes getting IPs from the policy.
+- As a cluster administrator, I want each of these static IPs to be unique, and taken from a defined range, and freed when unused.
+
+### Goals
+
+- Allow users to use NodeNetworkConfigurationPolicy to configure static cluster-unique IP addresses without hard-coding them.
+- If at all possible, we should pick a solution that could be utilized even in the IPI nmstate API.
+
+### Non-Goals
+
+- This design does not cover IP address configuration of management NIC on day-0/1
+
+## Proposal
+
+### Workflow Description (assign static address with IPPool)
+
+**cluster admin** is a human user responsible for managing the cluster node
+networking.
+
+1. The cluster admin create `NodeNetworkIPPool` resources with IP ranges and a names.
+2. The cluster admin configure node network interfaces statically but with IP from the `NodeNetworkIPPool` using a
+`NodeNetworkConfigurationPolicy` referencing them.
+3. The cluster admin wait for the `NodeNetworkConfigurationPolicy` to success
+
+#### Variation (not free IP address at the NodeNetworkIPPool)
+
+If there is not a free IP address at referenced pool NNCP will fail with a explicit message, so the range has to be
+increased and NNCP applied again
+
+### Workflow Description (removing a reference to NodeNetworkIPPool)
+
+**cluster admin** is a human user responsible for managing the cluster node
+networking.
+
+1. The cluster admin changes network interface to remove the refernece to the NodeNetworkIPPool (for example harcoding the IP or switching to DHCP).
+2. The cluster admin wait for the NodeNetworkConfigurationPolicy to success so the IPs from that interfaces are free now.
+
+### Workflow Description (scaling up cluster)
+
+**cluster admin** is a human user responsible for managing the cluster node
+networking.
+
+1. The cluster admin add a new node to the cluster already configured using IPools
+2. The cluster admin wait for the NodeNetworkConfigurationPolicy to success so the new node has took address from the NodeNetworkIPPool
+
+
+### Workflow Description (scaling down cluster)
+
+**cluster admin** is a human user responsible for managing the cluster node
+networking.
+
+1. The cluster admin remove a node form a cluster already configured using `NodeNetworkIPool`
+2. The cluster admin wait for the `NodeNetworkConfigurationPolicy` to success so the IPs from removed node are free.
+
+### Workflow Description (manual NodeNetworkIPPool clean up)
+
+**cluster admin** is a human user responsible for managing the cluster node
+networking.
+
+1. The cluster admin find that some IPs are not used anymore but still allocated
+2. The cluster admin reset `NodeNetworkIPPool` allocated status
```

Comment:
That's good point, we can Reconcile on status but we have to be carefull not to enter an infinite loop since the status change to change the ippool will also update the status conditions. 

Maybe an annotation is just fine ? At the end this is an emergency situation and in the future it maybe automatize.

What do you think ?